haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,308 @@
+from abc import ABC, abstractmethod
+import json
+import logging
+from typing import Any, Optional, Dict, Type, Union, List
+
+
+from haystack.errors import AWSConfigurationError, AmazonBedrockConfigurationError, AmazonBedrockInferenceError
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer.aws_base import AWSBaseInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import DefaultPromptHandler
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install farm-haystack[aws]'") as boto3_import:
+    from botocore.exceptions import ClientError
+
+
+class BedrockModelAdapter(ABC):
+    def __init__(self, model_kwargs: Dict[str, Any], max_length: Optional[int]) -> None:
+        self.model_kwargs = model_kwargs
+        self.max_length = max_length
+
+    @abstractmethod
+    def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
+        pass
+
+    @abstractmethod
+    def get_responses(self, response_body: Dict[str, Any]) -> List[str]:
+        pass
+
+    def _get_params(self, inference_kwargs: Dict[str, Any], default_params: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Merges the default params with the inference kwargs and model kwargs.
+
+        Includes param if it's in kwargs or its default is not None (i.e. it is actually defined).
+        """
+        kwargs = self.model_kwargs.copy()
+        kwargs.update(inference_kwargs)
+        return {
+            param: kwargs.get(param, default)
+            for param, default in default_params.items()
+            if param in kwargs or default is not None
+        }
+
+
+class AnthropicModelAdapter(BedrockModelAdapter):
+    def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
+        default_params = {
+            "max_tokens_to_sample": self.max_length,
+            "stop_sequences": ["\n\nHuman:"],
+            "temperature": None,
+            "top_p": None,
+            "top_k": None,
+        }
+        params = self._get_params(inference_kwargs, default_params)
+
+        body = {"prompt": f"\n\nHuman: {prompt}\n\nAssistant:", **params}
+        return body
+
+    def get_responses(self, response_body: Dict[str, Any]) -> List[str]:
+        responses = [response_body["completion"].lstrip()]
+        return responses
+
+    def get_stream_responses(self, response_body: Dict[str, Any]) -> List[str]:
```

Comment:
A couple of things that can be improved here:
- correct the type of `response_body`: `botocore.eventstream.EventStream`
- make `get_stream_responses an abstract method in `BedrockModelAdapter`, `AI21ModelAdapter` will implement it by just raising a `NotImplementedException`
- please don't use single character variables, `chunk` would be an appropriate name here
- the usage of the lambda does not really improve the readability here, instead consider this
  ```python
      for chunk in response_body:
          response_json = json.loads(chunk["chunk"]["bytes"].decode("utf-8"))
          response = response_json["completion"].strip()
          responses.append(response)
  ```